torch/fx/symbolic_trace.py

```diff
@@ -47,6 +47,24 @@ class Tracer(TracerBase):
     def __init__(self):
         super().__init__()
 
+    def call_module(self, m : torch.nn.Module, target : str, args, kwargs):
+        """
+        Method to override behavior when calling a module. By default, this
+        method checks `self.is_leaf_module()`. If `m` is not a leaf module,
+        trace through the original call using the `orig_call` method, preserved
+        from nn.Module.__call__. If it is a leaf module, emit a `call_module`
+        node.
+
+        You can override this method to get more sophisticated behavior.
+        For example, you might trace an entirely new GraphModule for the
+        submodule call, then emit a `call_module` node. This can be a way to
+        preserve the original module hierarchy.
+        """
+        if not self.is_leaf_module(m):
+            return m(*args, **kwargs)
```

Comment:
can't we just unpatch the `Module.__call__` before calling `Delegate.call_module` and here add something that patches it back:
```
with EmbedInline():
  m(*args, **kwargs)
```

or

```
embed_inline(m)(*args, **kwargs)
```

Reply:
This method is overridable by the caller and we want to avoid them having to know the implementation details of how our weird module call patching thing works if they want to create their own implementation

Reply:
In particular, it would be surprising if the user has a `Module` `Foo`, they see that `isinstance(m, Foo)`, so it would make sense that an instance of their own module would behave like they expect it to (i.e. `m(*args, **kwargs)` calls the module, not something else)